src/test/scala/org/apache/spark/sql/delta/UpdateSuiteBase.scala

```diff
@@ -0,0 +1,285 @@
+
+      checkUpdate(condition = Some("key >= 1"), setClauses = "value = key + value, key = key + 1",
+        expectedResults = Row(0, 3) :: Row(2, 5) :: Row(2, 2) :: Row(3, 4) :: Nil)
+    }
+  }
+
+  Seq(true, false).foreach { isPartitioned =>
+    test(s"basic update - with where and no row matched - Partition=$isPartitioned") {
+      val partitions = if (isPartitioned) "key" :: Nil else Nil
+      append(Seq((2, 2), (1, 4), (1, 1), (0, 3)).toDF("key", "value"), partitions)
+
+      checkUpdate(condition = Some("key >= 10"), setClauses = "value = key + value, key = key + 1",
+        expectedResults = Row(0, 3) :: Row(1, 1) :: Row(1, 4) :: Row(2, 2) :: Nil)
+    }
+  }
+
+  Seq(true, false).foreach { isPartitioned =>
+    test(s"type mismatch - Partition=$isPartitioned") {
+      val partitions = if (isPartitioned) "key" :: Nil else Nil
+      append(Seq((2, 2), (1, 4), (1, 1), (0, 3)).toDF("key", "value"), partitions)
+
+      checkUpdate(condition = Some("key >= 1"),
+        setClauses = "value = key + cast(value as String), key = key + '1'",
+        expectedResults = Row(0, 3) :: Row(2, 5) :: Row(3, 4) :: Row(2, 2) :: Nil)
+    }
+  }
+
+  Seq(true, false).foreach { isPartitioned =>
+    test(s"set to null - Partition=$isPartitioned") {
+      val partitions = if (isPartitioned) "key" :: Nil else Nil
+      append(Seq((2, 2), (1, 4), (1, 1), (0, 3)).toDF("key", "value"), partitions)
+
+      checkUpdate(condition = Some("key >= 1"),
+        setClauses = "value = key, key = null + '1'",
+        expectedResults = Row(0, 3) :: Row(null, 1) :: Row(null, 1) :: Row(null, 2) :: Nil)
+    }
+  }
+
+  Seq(true, false).foreach { isPartitioned =>
+    test(s"basic update - TypeCoercion twice - Partition=$isPartitioned") {
+      val partitions = if (isPartitioned) "key" :: Nil else Nil
+      append(Seq((99, 2), (100, 4), (101, 3)).toDF("key", "value"), partitions)
+
+      checkUpdate(
+        condition = Some("cast(key as long) * cast('1.0' as decimal(38, 18)) > 100"),
+        setClauses = "value = -3",
+        expectedResults = Row(100, 4) :: Row(101, -3) :: Row(99, 2) :: Nil)
+    }
+  }
+
+  test("Negative case - non-delta target") {
+    withTable("nondeltaTable") {
+      Seq((1, 1), (0, 3), (1, 5)).toDF("key1", "value").write.saveAsTable("nondeltaTable")
+      val e = intercept[AnalysisException] {
+        executeUpdate(target = "nondeltaTable", set = "key1 = 3")
+      }.getMessage
+      assert(e.contains("UPDATE destination only supports Delta sources"))
+    }
+  }
+
+  test("update cached table") {
+    withTable("cached_deltaTable") {
+      Seq((2, 2), (1, 4)).toDF("key", "value")
+        .write.format("delta").saveAsTable("cached_deltaTable")
+
+      spark.table("cached_deltaTable").cache()
+      spark.table("cached_deltaTable").collect()
+
+      executeUpdate(target = "cached_deltaTable", set = "key = 3")
+      checkAnswer(spark.table("cached_deltaTable"), Row(3, 2) :: Row(3, 4) :: Nil)
+    }
+  }
```

Comment:
we dont yet support subqueries right? So maybe add a test to verify that. similar to what you added in delete

Reply:
yes, on my way adding tests.

Reply:
done.